ironfish-rust/build.rs

```diff
@@ -0,0 +1,125 @@
+use sha2::Digest;
+use sha2::Sha512;
+use std::env;
+use std::fs;
+use std::io::BufRead;
+use std::io;
+use std::path::Path;
+use std::path::PathBuf;
+
+fn copy_if_present<P: AsRef<Path>, Q: AsRef<Path>>(src: P, dst: Q) {
+    let src = src.as_ref();
+    let dst = dst.as_ref();
+    if let Err(err) = fs::copy(src, dst) {
+        match err.kind() {
+            // If the source file is not found, it's fine to keep going: `fetch_if_missing` will
+            // download it later
+            io::ErrorKind::NotFound => (),
+            // All other errors should block the build
+            err => panic!("failed to copy {} to {}: {}", src.display(), dst.display(), err),
+        }
+    }
+}
+
+fn fetch_if_missing<P: AsRef<Path>>(dst: P) {
+    let dst = dst.as_ref();
+    if dst.exists() {
+        return;
+    }
+
+    let name = dst.file_name()
+                  .unwrap()
+                  .to_str()
+                  .unwrap();
+
+    println!("cargo:warning=fetching {name} from GitHub");
```

Comment:
I'm reconsidering showing this as a warning: this would be displayed for every build on every crate that depends on `ironfish`. Not a great user experience.

Reply:
I think it's good enough for now, but I agree, it's not a great permanent solution.

I think a good eventual state would be maybe some feature flags (and no default). I'm assuming feature-flags can be utilized in the build script, but that might not be the case - this is pretty uncharted territory for me. 

Maybe something like
 - `generate_params` which would generate params, good enough for testing or anyone who is using the crate and doesn't plan on connecting to our testnet/mainnet. This is a chicken-and-egg problem though, since you need the built library to create the parameters. Well, I guess technically, you could have `ironfish-zkp` as a build-dependency, since that's where the code lives for generating basic params (no trusted setup or anything fancy): https://github.com/iron-fish/ironfish/blob/master/ironfish-zkp/src/bin/generate_params.rs
 - `download_params` which would download the params if missing/invalid hash or whatever.
 - Maybe even a third that is like `no_params` which requires you to bring your own.

Just kind of brainstorming here, I don't think any of these changes should be added in this PR.

Reply:
Fully agree. And yes, `$[cfg(feature = ...)]` are usable from the build scripts, so this is perfectly doable.